client/packages/executor/src/executionManager/sideEffect.ts

```diff
@@ -178,10 +206,57 @@ export class SideEffect extends EventEmitter {
             this.insurance = sdk.circuit.toFloat(sideEffect.insurance) // this is always in TRN (native asset)
             this.strategyEngine = strategyEngine
             this.biddingEngine = biddingEngine
-            this.circuitSignerAddress = circuitSignerAddress
             this.logger = logger
-            this.vendor = this.gateway.vendor;
+            this.vendor = this.gateway.vendor
+        }
+    }
+
+    /** Custom JSON serialization. */
+    toJSON(): SerializableSideEffect {
+        return {
+            id: this.id,
+            xtxId: this.xtxId,
+            sideEffect: Object.assign({}, this.rawSideEffect, { action: this.rawSideEffect.action.toHuman() }),
+            misc: {
+                executorName: this.misc.executorName,
+                logsDir: this.logger.logsDir,
+                circuitRpc: this.misc.circuitRpc,
+                circuitSignerAddress: this.misc.circuitSignerAddress,
+                circuitSignerSecret: this.misc.circuitSignerSecret,
+                gatewayId: this.gateway.id,
+                xdns: Object.assign({}, this.gateway.record),
+            },
+        }
+    }
+
+    /** Custom JSON deserialization. */
+    static fromJSON(o: SerializableSideEffect): SideEffect {
+        const logger = createLogger(o.misc.executorName, o.misc.logsDir)
+        const sdk = new Sdk(o.misc.circuitRpc, o.misc.circuitSignerSecret)
+        let sideEffect = {
+            ...o.sideEffect,
+            action: {
+                toHuman() {
+                    return o.sideEffect.action.toString()
+                },
+            },
+            target: {
+                toU8a() {
+                    return Uint8Array.from(Buffer.from(o.misc.gatewayId))
+                },
+            },
         }
+        sideEffect = new Gateway(o.misc.xdns).createSfx[o.sideEffect.action](sideEffect) as T3rnTypesSideEffect
```

Comment:
We can use the sdk instance here directly, so `sdk.gateways[gatewayId].createSfx[o.sideEffect.action](sideEffect)`

Reply:
The problem is that initializing the sdk (i.e. initializing its gateways) is not feasible at this point because we are in the prescribed sync fromJSON, whose signature we do not control, whereas `sdk.init` is async.